configure.ac

```diff
@@ -551,10 +547,11 @@ AM_CONDITIONAL(HAVE_SYSTEMD, [test -n "$with_systemdsystemunitdir" -a "x$with_sy
 AC_ARG_WITH([systemduserunitdir],
         AS_HELP_STRING([--with-systemduserunitdir=DIR], [Directory for systemd user service files]),
         [], [with_systemduserunitdir=$($PKG_CONFIG --variable=systemduserunitdir systemd)])
-if test "x$with_systemduserunitdir" != xno; then
+AM_CONDITIONAL([USE_LIBSYSTEMD], [test -n "$have_libsystemd" -a -n "$with_systemduserunitdir" -a "x$with_systemduserunitdir" != xno])
+if test -n "$have_libsystemd" -a -n "$with_systemduserunitdir" -a "x$with_systemduserunitdir" != xno ; then
         AC_SUBST([systemduserunitdir], [$with_systemduserunitdir])
+        AC_DEFINE([USE_LIBSYSTEMD], [1], [Used for activation via socket])
```

Comment:
Rather call AM_CONDITIONAL after the if, using the USE_LIBSYSTEMD variable in the test. Otherwise sooner or later, somebody will modify line 550 without thinking about modifying line 551, and we'll get incoherent

Reply:
I should have mentioned that I tried that but autoconf complains.

>  Note that you must arrange for every AM_CONDITIONAL to be invoked every time configure is run. If AM_CONDITIONAL is run conditionally (e.g., in a shell if statement), then the result will confuse automake. 

https://www.gnu.org/software/automake/manual/html_node/Usage-of-Conditionals.html

Not sure what is the best way to circumvent that?

Reply:
I'm not saying to put it inside the if, but *after* it, and use the USE_LIBSYSTEMD variable to know whether the if succeeded or not.

Reply:
Please don't merge yet, I must have be misunderstanding, because
```
AM_CONDITIONAL([USE_LIBSYSTEMD], [USE_LIBSYSTEMD])
```
wouldn't work.

Reply:
See how it's done with other, here it's be `[test x$USE_LIBSYSTEMD = x1]`

Reply:
Oh, no, sorry, AC_DEFINE doesn't actually define an environment variable. Then define one.

Reply:
Thanks for your help! I tested and works as intended with/without `libsystemd-dev` on the host.